src/GitHub.Api/SimpleApiClientFactory.cs

```diff
@@ -28,10 +28,6 @@ public SimpleApiClientFactory(IProgram program, Lazy<IEnterpriseProbeTask> enter
 
         public ISimpleApiClient Create(UriString repositoryUrl)
         {
-            var contains = cache.ContainsKey(repositoryUrl);
-            if (contains)
-                return cache[repositoryUrl];
-
```

Comment:
I'm almost certain I did this because Create will never get called at the same time as ClearFromCache, and the lock in Create is not to protect Create from ClearFromCache, but to have only the first instance that comes into Create trigger the creation, and keep everyone else from never even reaching the lock.

When the team explorer home page loads, there's a series of requests that get triggered to go through Create, all at the same time - one for each section and navigation item. If there's already an api client, great, return that, but if there isn't, whoever gets to the lock first gets to create it. Since the section gets created first and there's a slight delay between it and the nav items being created, that means that potentially the nav items never even hit the lock and just return immediately.

I need to confirm this, and maybe change the names on these locks to be more clear about what their role is.